### PR TITLE
pdpdevtool-5579: oauth2 client credentials flow

### PR DESCRIPTION
**Pull request by davidecorreu**

### CODE REVIEW

**Review by antoniofruci — APPROVED** [review]

File: packages/node-cli/src/commands/account/setupci/AccountSetupCiAction.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+			params[COMMAND.OPTIONS.ACCOUNT] = params[COMMAND.OPTIONS.ACCOUNT].toUpperCase();
+		}
+		return params;
+	}
+
+	async execute(params) {
+		if (params[COMMAND.OPTIONS.DOMAIN] === GENERIC_NETSUITE_DOMAIN) {
+			delete params[COMMAND.OPTIONS.DOMAIN];
+		}
+		return await auhtenticateCi(params, this._sdkPath, this._executionPath, this._executionEnvironmentContext);
```
Comment (lfpvillegas) [inline packages/node-cli/src/commands/account/setupci/AccountSetupCiAction.js:50]: typo

File: packages/node-cli/src/commands/account/setupci/AccountSetupCiAction.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+	}
+
+	_checkWorkingDirectoryContainsValidProject() {
```
Comment (lfpvillegas) [inline packages/node-cli/src/commands/account/setupci/AccountSetupCiAction.js:54]: I think this same method is used in several classes. Could we move it to BaseAction?

File: packages/node-cli/src/utils/AuthenticationUtils.js [hunk 3]
```diff
@@ -153,4 +164,35 @@ async function authenticateWithOauth(params, sdkPath, projectFolder, cancelToken
 		.catch((error) => AuthenticateActionResult.Builder.withErrors([error]));
 }
 
-module.exports = { setDefaultAuthentication, getProjectDefaultAuthId, getAuthIds, saveToken, authenticateWithOauth };
+async function auhtenticateCi(params, sdkPath, projectFolder, executionEnvironmentContext) {
```
Comment (lfpvillegas) [inline packages/node-cli/src/utils/AuthenticationUtils.js:167]: typo

File: packages/node-cli/src/metadata/SdkCommandsMetadata.json [hunk 4]
```diff
@@ -58,6 +58,65 @@
 			}
 		}
 	},
+	"account:setup:ci": {
+		"name": "account:setup:ci",
+		"sdkCommand": "authenticateci",
+		"usage": "",
+		"description": "Grants authorization for SuiteCloud SDK to access NetSuite with an account-role combination. It does not require browser-based login to NetSuite.",
```
[the remark this answers is not in the document]
Comment (davidecorreu) [reply packages/node-cli/src/metadata/SdkCommandsMetadata.json:65]: It should be selected when uploading the public part of the certificate into the account.

File: packages/node-cli/src/commands/account/setupci/AccountSetupCiAction.js (hunk 2 above)
Comment (davidecorreu) [reply packages/node-cli/src/commands/account/setupci/AccountSetupCiAction.js:54]: I'll check and I let you know.

File: packages/node-cli/src/metadata/SdkCommandsMetadata.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (lfpvillegas) [reply packages/node-cli/src/metadata/SdkCommandsMetadata.json:65]: That's true. Let's resume the conversation outside

**Review by lfpvillegas — APPROVED** [review]

**Pull request merged**